internal/common/conversion/schema_description.go

```diff
@@ -0,0 +1,80 @@
+package conversion
+
+import (
+	"reflect"
+
+	dsschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+func UpdateSchemaDescription(s *schema.Schema) {
+	updateAttr(s)
+}
+
+func UpdateDSSchemaDescription(s *dsschema.Schema) {
+	updateAttr(s)
+}
+
+func updateAttr(attr any) {
+	ptr := reflect.ValueOf(attr)
+	if ptr.Kind() != reflect.Ptr {
+		panic("not ptr, please fix caller")
+	}
+	v := ptr.Elem()
+	if v.Kind() != reflect.Struct {
+		panic("not struct, please fix caller")
```

Comment:
added multiple panic to make sure it works as expected. 

These panics will be caught in unit tests so they won't make it to master, e.g. in TestResourceSchemas